docs/docs/deploying-to-s3-cloudfront.md

```diff
@@ -54,6 +54,25 @@ And finally, add the deployment script to your `package.json`:
 That's it!
 Run `npm run build && npm run deploy` to do a build and have it immediately deployed to S3!
 
+## Deploying with `.env` variables
+
+Some deployments require passing environment variables. To deploy with environment variables, update the deployment script to your `package.json`:
+
+```js:title=package.json
+"scripts" : {
+    ...
+    "deploy": "npm run -n \"-r dotenv/config\" && npm run build && gatsby-plugin-s3 deploy"
+}
+```
+
+This command requires `dotenv` first, runs build next, and finally deploys to s3. `dotenv`, already a dependency of Gatsby, loads environment variables and makes it available globally.
```

Comment:
```suggestion
This command requires `dotenv` first, runs build next, and finally deploys to s3. `dotenv`, a dependency of Gatsby that doesn't need to be installed directly, loads environment variables and makes them available globally.
```

Reply:
thank you for your patience with me! I pushed changes :)

Reply:
Thank _you_! This looks great!